src/machine/builder.rs

```diff
@@ -0,0 +1,44 @@
+use std::io::{Stdout, Write};
+
+use crate::machine::{
+    arithmetic::{Arithmetic, OverflowArithmetic},
+    env::Env,
+    Machine,
+};
+
+pub struct MachineBuilder<W: Write, A: Arithmetic> {
+    env: Env<W>,
+    arithmetic: A,
+}
+
+impl Default for MachineBuilder<Stdout, OverflowArithmetic> {
+    fn default() -> Self {
+        MachineBuilder {
+            env: Env::default(),
+            arithmetic: OverflowArithmetic,
+        }
+    }
+}
+
+impl<W: Write, A: Arithmetic> MachineBuilder<W, A> {
+    pub fn build(self) -> Machine<W, A> {
+        Machine {
+            env: self.env,
+            arithmetic: self.arithmetic,
+        }
+    }
+
+    pub fn with_env<W2: Write>(self, env: Env<W2>) -> MachineBuilder<W2, A> {
+        MachineBuilder {
+            env,
+            arithmetic: self.arithmetic,
```

Comment:
Does `..self` work here? Then we wouldn't need to change these methods when we decide to add more fields.

Reply:
sadly no because the new builder and the old one have different types :/